sqlglot/dialects/tsql.py

```diff
@@ -42,11 +51,28 @@ def _format_time(args):
     return _format_time
 
 
+def parse_format(args):
+    fmt = list_get(args, 1)
+    number_fmt = not DATE_FMT_RE.search(fmt.this) or fmt.name in TRANSPILE_SAFE_NUMBER_FMT
+    if number_fmt:
+        return exp.NumberToStr(this=list_get(args, 0), format=fmt)
+    return exp.TimeToStr(this=list_get(args, 0), format=exp.Literal.string(format_time(fmt.name, TSQL.time_mapping)))
+
+
 def generate_date_delta_with_unit_sql(self, e):
     func = "DATEADD" if isinstance(e, exp.DateAdd) else "DATEDIFF"
     return f"{func}({self.format_args(e.text('unit'), e.expression, e.this)})"
 
 
+def generate_format_sql(self, e):
+    fmt = (
+        e.args["format"]
+        if isinstance(e, exp.NumberToStr)
+        else exp.Literal.string(format_time(e.args.get("format").name, TSQL.inversed_lower_case_time_mapping))
```

Comment:
why would you want a lower case mapping? what's the point of that since time_mapping already has lower case stuff? don't you only want an inverse mapping? additionally, sqlglot automatically produces an inverse_time_mapping, so you don't need to remake it

Reply:
tsl doesn't allow YYYY for example in FORMAT

Reply:
Technically it will allow it but for example FORMAT('01-01-1990', dd.mm.YYYY) => 01.01.YYYY which is not the intended output. Better solution might be to force somehow the format_time to return the lowercase by default but I dont think other dialects have any benefit of that.

Reply:
just do the lower case afterwards intsead of in the mapping

Reply:
Yea you are right it doesn't change anything the way it is here. But I should have checked for YYYY in the dict comprehension. MM and mm are different formaters, so pushing all to lower is wrong.